Gds/src/fprime_gds/common/utils/string_util.py

```diff
@@ -0,0 +1,44 @@
+"""
+string_util.py
+
+Utility functions to process strings to be used in FPrime GDS
+
+@Created March 18, 2021
+@janamian
+"""
+
+import re
+
+def format_string(format_str, values):
+    """
+    Function to convert C-string style to python format
+    without using python interpolation
+    Considered the following format for C-string:
+    %[flags][width][.precision][length]type
+
+    %:           (?<!%)(?:%%)*%
+    flags:       ([\-\+0\ \#])?
+    width:       (\d+|\*)?
+    .precision:  (\.\*|\.\d+)?
+    length:      ([hLIw]|l{1,2}|I32|I64)?
+    type:        ([cCdiouxXeEfgGaAnpsSZ])
```

Comment:
Personally, instead of adding the items you've added to `expect`, I'd consider adding one of the following to https://github.com/nasa/fprime/blob/devel/.github/actions/spelling/patterns.txt

```
# See Gds/src/fprime_gds/common/utils/string_util.py
\[cCdiouxXeEfgGaAnpsSZ\]
```
This would cause just this strange single instance (or any identical instance) to be ignored.

```
# See Gds/src/fprime_gds/common/utils/string_util.py
^\s*type:\s*\(\[[^]*\]\)$
```
This should cause a line like this to be ignored without explicitly managing the precise contents of the file.

Reply:
@jsoref, thank you for the comment.
I agree we should not keep adding sequences to the expect.txt for spell checking.
I also think adding specific patterns is not a better strategy. For example in the case above `LIw` is one of the sequences that is getting confuse by the spell checker which is on line 23.
In my opinion we need to define opening and closing pattern where the spell checker should not need to check for spelling. I would suggest using back-tick to hint the spell checker to not check for spelling. 
We can add the following pattern to skip any text between two back-ticks:
https://github.com/nasa/fprime/blob/devel/.github/actions/spelling/patterns.txt
```
`.*?`
```
We also need to add a guideline page for contributors to know how to use these techniques.
We should also reduce noises created by the spell checker since spelling, although important for readability, is not the main purpose of most pull requests.

Reply:
@jsoref, I added two patterns to escape the inline back-ticks and in-code sequences with special symbols. I was not able to add multiline escape pattern as for some reason spell checker ignores `\m` (multiline flag). The prefer method would be to be able to ignore anything between three back-tick block in the comments similar to the following:

/*
Text to be spell checked

\`\`\` <-------- block start
Text to not be spell checked
\`\`\` <-------- block end

Text to be spell checked
*/

FYI: This is the regex I tried to use for multiline block:
```regex
^[ \t]*`{3}[^\n]*[\s\S]+?^[ \t]*`{3}/m
```

Reply:
Unfortunately I don't have a multi line notation. You could define something like `no-spell-check`

And then have this in `patterns.txt`:
```
^.*no-spell-check.*$
```

You can add a file `advice.txt` next to `patterns.txt` suggesting people add `// no-spell-check` to lines to skip them.

Reply:
Yeah, thinking about multiline patterns is on my list. Currently I read files line at a time. Projects by default check all files, likely including large binary files. I've been a bit worried about forcing it to handle large files as a whole.

I'm currently working on teaching the tool to recognize binary files, but amusingly the last thing I looked at was an `.exe` and the tool rightly said _there are **real misspellings here**, and there are **fewer misspelled items** than **correctly spelled ones**_. (Arguably still the wrong behavior since it'd be better to catch those misspellings at their source... But amusing.) I think the heuristic I wrote should at least guard against non English content even if it fails on "binary" content. I'm thinking about trying a file size vs content heuristic which should hopefully catch the `.exe` case.

Once I've added a heuristic that lets the tool gracefully adjust for binary files, I think I'll be more comfortable trying to handle multiline matching. Although I'm still leaning towards using a distinct file for it.